streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -406,6 +407,10 @@ private void maybePunctuate(StreamTask task) {
             // which are essentially based on record timestamp.
             if (task.maybePunctuate())
                 sensors.punctuateTimeSensor.record(time.milliseconds() - now);
+            final ProcessorNode node = task.node();
+            if (node != null) {
+                node.nodeMetrics.nodePunctuateTimeSensor.record();
```

Comment:
Is it correct to do the punctuation sensor recording for task.node() or even here? Why not do it in `StreamTask's` `punctuate`?